File: backend/tests/test_data/open_validation/benchmark_cox.py
import pandas as pd
from lifelines import CoxPHFitter
from lifelines.statistics import proportional_hazard_test

# 1. Load Data
df = pd.read_csv("../benchmark_cox.csv")

# 2. Fit Cox Model
cph = CoxPHFitter()
cph.fit(df, duration_col='week', event_col='arrest')

# 3. Output Results
print(cph.summary)

# 4. Check Assumption
print("PH Assumption Check:")
results = proportional_hazard_test(cph, df, time_transform='rank')
print(results.print_summary())

File: backend/tests/test_data/open_validation/benchmark_cox.R
# benchmark_cox.R
# Insight Platform - Open Validation Pack
# Goal: Verify Cox Proportional Hazards Model

# 1. Load Libraries
if (!require("survival")) install.packages("survival")
library(survival)

# 2. Load Data
df <- read.csv("../benchmark_cox.csv")

# 3. Fit Cox Model
# Using Efron handling for ties (Standard)
model <- coxph(Surv(week, arrest) ~ fin + age + prio, data = df, ties = "efron")

# 4. Output Results
print(summary(model))
print("Coefficients:")
print(coef(model))
print("HR (Exp Coef):")
print(exp(coef(model)))

# 5. Check Assumptions
test.ph <- cox.zph(model)
print("PH Assumption Check:")
print(test.ph)
